doc/src/manual/types.md

```diff
@@ -531,8 +531,7 @@ ERROR: TypeError: in typeassert, expected Union{Int64, AbstractString}, got a va
 
 The compilers for many languages have an internal union construct for reasoning about types; Julia
 simply exposes it to the programmer. The Julia compiler is able to generate efficient code in the
-presence of `Union` types with a small number of types [^1], by generating specialized code
-in separate branches for each possible type.
+presence of `Union` types, by generating specialized code in separate branches for each possible type.
```

Comment:
While the limit has been lifted for specific situations, as you discovered it's not completely gone. Maybe best to acknowledge there are still limits?